shared-articles/what does dcdd do.md

```diff
@@ -0,0 +1,90 @@
+
+Therefore, since the network participants are each following the same set of rules, and everyone is sharing the same ledger between each other, it's impossible to go back and retroactively alter the ledger. You can only add new information to the ledger. That's what makes old blocks immutable.
+
+The software that makes this all possible on the Decred blockchain is dcrd. dcrd is written in Go (golang), and implements all the consensus rules and core blockchain ledger functionality. It is important to note that dcrd does NOT include wallet functionality, and was designed that way for security reasons. You can consider dcrd to be your personal Decred blockchain server, which would connect you to the rest of the Decred network.
+
+## dcrd responsibilities:
+dcrd is the backbone of Decred's peer-to-peer network. dcrd acts as a 'full node', short for 'fully-validating node', which means that it fully validates all transactions and blocks, as opposed to trusting a 3rd party or someone else's ledger. In summary, a dcrd full node:
+
+* Fully validates all transactions and blocks (by enforcing the Decred consensus rules),
+* Maintains the entire past transactional ledger of Decred,
+* Relays newly mined blocks,
+* Gives the ability to mine blocks,
+* Allows relaying of transactions to other Decred nodes around the world (by sharing a transaction mempool),
+* Allows sharing a list of peers in the Decred network so that each new peer can talk to as many other peers as possible.
+
+In terms of benefits, since dcrd fully validates every block and transaction, it provides the highest security and privacy possible when used with a wallet that’s also in full validation mode. This is ideal for individuals, businesses, and services that need the most reliable and accurate data about transactions.
+
+## Full Node Distinctions
+As Decred builds out its infrastructure, it becomes apparent that the underlying security is a high priority. Each time a new service is added, it increases the opportunity for node participation. This is done on multiple levels, and it’s important to note that not every node is equal. There are two factors to consider when determining the usefulness of a node:
+
+**Uptime**
+* High-uptime (running 24/7/365, or close to it)
+* Low-uptime (running intermittently)
+**With inbound connections**
+* No (default)
+* Yes (requires the user to open up a port in their firewall)
+
+**Dcrd configuration options include**
+* High-uptime with inbound connections
+* High-uptime without inbound connections
+* Low-uptime with inbound connections
+* Low-uptime without inbound connections
+
+The Full node that supports the network the best is the High-uptime with inbound connections.
+
```

Comment:
Let's flesh out WHY people would be incentivized to allow inbound connections?

---
## Incentives

What would incentivize someone to run a dcrd server with high-uptime and inbound connections? There's maintenance hassle, hardware and electricity costs, and some inherent security risk by exposing a port in their network firewall so that dcrd can have inbound connections from the public internet.

This is one of the many differences between Bitcoin and Decred. In the Bitcoin network, there's no real incentive for people to run their own full nodes, other than to not have to trust a 3rd party. People who run Bitcoin nodes are doing it for fun or their own goodwill. However, with Decred, people are incentivized to run full nodes so that they can solo-stake so that they can earn the staking rewards.

Note, since many people aren't technical enough to run their own dcrd servers, but still want to participate in staking, Decred created Voting Service Providers (VSPs), which will be talked about later. VSPs allow staking, but  don't require people to run a server or mess with their firewalls.

### Uptime

Solo-staking requires people to be actively watching the chain and validating new blocks. Therefore, high-uptime is important if you don't want to miss a vote!

### Connections

dcrd is hardcoded to only ever create [8 outbound connections](https://docs.decred.org/faq/configuration/#5-why-am-i-connecting-to-only-8-outbound-peers). By default, dcrd allows 125 inbound connections, but the server operator needs to open up a hole in their firewall for any of those inbound connections to be used.

What's the difference between inbound and outbound connections? If your node initiated the connection to an external peer, it's outbound. Otherwise, it's inbound. Outbound/inbound connections don't share different types of data. Nodes will send and receive data from both types of connections exactly the same way.

Ultimately, by allowing dcrd to have inbound connections, solo-stakers get to be connected to more peers (up to 125 more). This means they can receive messages propagating through the network faster, and they're also less vulnerable to things like eclipse or sybil attacks. Getting messages faster is important, especially if you want to get your vote included in the latest block.

Reply:
I like having separate sections about Uptime and Connections.

Let's see if this piece fits the size @PhoenixGreen is looking for. In any case I hope it gets published one way or another. You're good at writing up complex tech stuff.